### PR TITLE
Bootstrap database creation

### PR DESCRIPTION
**Pull request by thumbnail**
automatically creates `DATABASES` if they do not exists upon startup.

### CODE REVIEW

File: create-dbs.clj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+(require
+ '[datomic.api :as d]
+ '[clojure.string :as str])
+
+(doseq [name (-> (System/getenv "DATABASES")
+                 (str)
+                 (str/split #"\,"))]
+  (when (and (not (str/blank? name))
+             (d/create-database (str "datomic:dev://localhost:4334/" name)))
+    (println "Created database:" name)))
```
Comment (lennartbuit) [inline create-dbs.clj]: You could do something like this to verify that all database are created:

```Clojure
(let [uri      "datomic:dev://localhost:4334/"
      db-names (-> (System/getenv "DATABASES")
                   (str)
                   (str/split #"\,")
                   (remove str/blank?)
                   (set))]
  (doall (map (fn [name] (d/create-database (str uri name)))
              db-names))
  (when-let [dbs (seq (set/difference db-names (set (d/get-database-names uri))))]
    (throw (ex-info "Unable to create some databases" {:databases dbs}))))
```

Comment (lennartbuit) [reply create-dbs.clj]: Also, maybe put this in a `defn -main`, which should also lein run

File: create-dbs.clj [hunk 2]
```diff
@@ -0,0 +1,10 @@
+(require
```
Comment (lennartbuit) [inline create-dbs.clj]: This can be a 'regular' ns, no?

Comment (thumbnail) [reply create-dbs.clj]: Nope; it's a script read by datomic repl

File: create-dbs.clj (hunk 1 above)
Comment (thumbnail) [reply create-dbs.clj]: There's no lein involved, hence lack of main. I'll update the script though.

File: create-dbs.clj [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+(let [uri      "datomic:dev://localhost:4334/"
+      db-names (-> (System/getenv "DATABASES")
+                   (str)
+                   (str/split #"\,")
+                   (as-> names (remove str/blank? names))
+                   (set))]
+  (run! (fn [name] (d/create-database (str uri name "?password=123")))
+        db-names)
+  (when-let [dbs (seq (set/difference db-names (-> (str uri "?password=123") d/get-database-names set)))]
```
Comment (Gluestick) [inline create-dbs.clj]:
```suggestion
  (when-let [dbs (seq (set/difference db-names (-> (str uri "*?password=123") d/get-database-names set)))]
```

File: README.md [hunk 4]
```diff
@@ -17,18 +17,22 @@ You can run this as standalone docker container, or in a docker-compose configur
...
 
 ### `docker-compose.yml`
 ```yml
 datomic:
     image: quay.io/nedap/datomic:latest
     command: ["./bin/transactor", "transactor.properties"]
+    environment:
+      DATABASES: "example,another"
     ports:
       - 4334-4336
```
Comment (Gluestick) [inline README.md]: Should be bound ports, e.g. `4334:4334`, right?

Comment (thumbnail) [reply README.md]: Heavily dependent on the setup. If you want to use them from localhost as a replacement: sure. If you want to use it in a isolated test/network: no.

File: transactor.properties [hunk 5]
```diff
@@ -24,7 +27,9 @@ license-key=$DATOMIC_LICENSE
 
 # h2-port=4335
 # h2-web-port=4336
-
+storage-access=remote
+storage-datomic-password=123
+storage-admin-password=123
```
Comment (thumbnail) [inline transactor.properties]: These passwords are only used in development to allow the containers to talk with each other and are not used in any non-local deployment.

**Pull request merged**